### ISSUE TITLE
Option to override the Windows Search whenever we type in the Start Menu

### ISSUE THREAD

**Issue by MakeshVineeth** [adirh3/Fluent-Search#251]
It would be awesome if FS can overtake the search whenever we type in Start Menu. Provide it as an option.

Why? Since just clicking **Win** key and typing are way easier and many people got used to it. Also, it doesn't seem to require any admin permissions too.

**EverythingToolbar** has implemented it recently (which can be used as an example): https://github.com/stnkl/EverythingToolbar/releases/tag/0.7.1

**Comment by adirh3** [comment]
@MakeshVineeth Why not use Win key and Fluent Search hotkey?

**Comment by MakeshVineeth** [comment]
> @MakeshVineeth Why not use Win key and Fluent Search hotkey?

I think this way we don't need to sacrifice the Start Menu. Since FS can be a Window Search Replacement, but not the Start Menu itself. Moreover, doesn't the Win key method require admin permission?

**Comment by adirh3** [comment]
> > @MakeshVineeth Why not use Win key and Fluent Search hotkey?
> 
> I think this way we don't need to sacrifice the Start Menu. Since FS can be a Windows Search Replacement, but not the Start Menu itself. Moreover, doesn't the Win key method require admin permission?

Admin permissions is only required when the focused app is running as admin, like Task Manager. 
The Start Menu is running as admin as well so to make Fluent Search pop while typing it will be problematic as well.
Although this might be possible with Instant Search + admin permissions.

**Comment by MakeshVineeth** [comment]
> > > @MakeshVineeth Why not use Win key and Fluent Search hotkey?
> > 
> > 
> > I think this way we don't need to sacrifice the Start Menu. Since FS can be a Windows Search Replacement, but not the Start Menu itself. Moreover, doesn't the Win key method require admin permission?
> 
> Admin permissions is only required when the focused app is running as admin, like Task Manager.
> The Start Menu is running as admin as well so to make Fluent Search pop while typing it will be problematic as well.
> Although this might be possible with Instant Search + admin permissions.

Either way, I think overriding the Windows Search can be convenient feature than disabling/sacrificing Start Menu altogether. Moreover, Users on Discord also seem to find this suggestion quite intuitive. Instead of Windows Search, when we type in the Start Menu, FS Search should pop up in the same position as Windows Search, I feel it will be so much easier and consistent that way.

**Comment by IgnusG** [comment]
I don't think this will be possible in windows 11 with the native search in the start menu (as discussed [here](https://github.com/stnkl/EverythingToolbar/issues/178) for everythingtoolbar). It might be possible to inject a search bar on the right similar to the traffic manager example from the linked discussion but it would be separate from the search bar on windows (so far everythingtoolbar also does not appear to have any traction on such implementation).

The `Win` key and start typing also won't be possible I think since it opens the start menu at which point all external apps loose control. 

As usual, there might be ways to do this in a very hacky roundabout patch-fix way but it would almost certainly break every time MS updates their internal API's around the new start menu and taskbar. Best to try and convince MS about exposing API's to allow more modification of the taskbar and start menu to third party apps (they probably won't do that though).

**Comment by IgnusG** [comment]
As a workaround, if you use [Windows Powertoys](https://github.com/microsoft/PowerToys) however you can remap either the <kbd>Win</kbd> key or the combination <kbd>Win</kbd> + <kbd>S</kbd> (by default search) to the shortcut you assigned to fluent search. So pressing them will open fluent search instead.

![image](https://user-images.githubusercontent.com/6438760/135441129-4dabe6db-579a-4e7f-b2fe-674094ef6ea9.png)

> Careful with remapping the windows key. You won't be able to use it for any other windows key shortcuts - so opening the start menu with it obviously won't work but other shortcuts which use it also won't work <kbd>Win</kbd> + <kbd>Tab</kbd> etc.

**Comment by adirh3** [comment]
@MakeshVineeth I believe the original suggestion was to make FS open with Win key, right? If so please close this

**Comment by MakeshVineeth** [comment]
The original suggestion was to launch FS **only** while typing into the Windows Search Bar (like in EverythingToolbar), that way we won't sacrifice the Start Menu while keeping the FS Search intact. But like @IgnusG has said, if this is something that is not possible through normal means, then this issue can be closed ;)
